app/controllers/matches_controller.rb

```diff
@@ -63,9 +64,40 @@ def match_params
     params.permit(:redAtt, :redDef, :blueAtt, :blueDef, :redGoal, :blueGoal, )
   end
 
-  private
   def update_players_for_match
     @match.add_players_for_match(match_params[:redAtt], match_params[:redDef], match_params[:blueAtt], match_params[:blueDef])
   end
 
+  def validation
+    @goal_check = true
+    @player_check = false
+
+    check_players([match_params[:redAtt], match_params[:redDef], match_params[:blueAtt], match_params[:blueDef]])
+    check_goal(match_params[:redGoal].to_i, match_params[:blueGoal].to_i)
+
+    error = {}
+    error[:goal] = false if @goal_check
+    error[:player] = false if @player_check
+    error
+  end
+
+  def check_goal (goal1, goal2)
+    @goal_check = false if goal1 == 10 && goal2 < 10
+    @goal_check = false if goal2 == 10 && goal1 < 10
+  end
+
+  def check_players (players)
```

Comment:
Do you want to check if each player is unique? Or do we consider the case where 1 person can play 2 roles? (eg. @sohymg plays red_atk and red_def :stuck_out_tongue:)